plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2801,7 +2801,7 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
                         Optional.empty()), // forAnalyze does not affect stats
                 handle -> {
                     Table icebergTable = catalog.loadTable(session, handle.getSchemaTableName());
-                    return TableStatisticsReader.getTableStatistics(typeManager, session, handle, icebergTable);
```

Comment:
@chenjian2664 `Fix table statistics collecting error with metadata column predicate` does the title of the commit still stand?

I've reverted your changes and didn't encounter an error in Trino/ but rather an assertion error in `testCollectingStatisticsWithPathColumnPredicate`